nixos/modules/services/networking/knot.nix

```diff
@@ -0,0 +1,93 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.knot;
+
+  configFile = pkgs.writeText "knot.conf" cfg.extraConfig;
+  socketFile = "/run/knot/knot.sock";
```

Comment:
Nitpick: thanks to `DynamicUser = "yes";` there won't be anything else in `/run/knot` AFAIK, so we might just call it `"/run/knot.sock"`, though the less-standard location might not be worth this simplification.